PROTOCOL.md

```diff
@@ -0,0 +1,231 @@
+# Protocol Specification
+
+The ttrpc protocol uses simple frames for communicating messages between a
+client and a server. The client represents the process which initiated the
+underlying connection and the server is the process which accepted the
+connection. The protocol is currently defined as asymmetrical, with clients
+sending requests and servers sending responses. Both clients and servers are
+able to send stream data. The roles are also used in determining the stream
+identifiers, with client initiated streams using odd number identifiers and
+server initiated using even number. The protocol may be extended in the future
+to support server initiated streams, that is not supported in the latest
+version.
+
+## Purpose
+
+The ttrpc protocol is designed to be lightweight and optimized for low latency
+and reliable connections between processes on the same host. The protocol does
+not include features for handling unreliable connections such as handshakes,
+resets, pings, or flow control. The protocol is designed to be easy to implement
```

Comment:
s/easy/light/

Reply:
The statement is intending to say that low overhead implementations are easy. Not sure about the wording and I can see that isn't exactly clear. Replacing easy with light doesn't convey that point though.

Reply:
"The protocol is designed to make low-overhead implementations as simple as possible"

Reply:
Updated